### PR TITLE
Add scss support for more powerful stylesheets

### PR DESCRIPTION
**Pull request by zwade**
Adds scss compilation on startup, as well as refactoring of css into scss

### CODE REVIEW

**Comment by zwade** [comment]
Resolves #28

File: scoreboard/logger.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ class Formatter(logging.Formatter):
 
     def format(self, record):
         if flask.request:
-            user = ('UID<%d>' % flask.g.uid) if flask.g.uid else '-'
+            user = ('UID<%d>' % flask.g.uid) if 'uid' in flask.g else '-'
```
Comment (k9delight) [inline scoreboard/logger.py]: Bug fixes belong in their own commits :)

Comment (k9delight) [inline scoreboard/logger.py]: Not sure if you've seen this, but git add -p makes it easy to add partial changes to commits, leaving the reminder to be committed/removed later on. More information at https://git-scm.com/book/en/v2/Git-Tools-Interactive-Staging

Comment (Matir) [inline scoreboard/logger.py]: They are also not sematically equivalent: e.g., if flask.g.uid = None

File: scoreboard/app.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 import logging
 import os
 from werkzeug import exceptions
+from flask_scss import Scss
```
Comment (Matir) [inline scoreboard/app.py]: Please import modules & packages but not individual classes.

File: scoreboard/app.py [hunk 3]
```diff
@@ -66,6 +67,10 @@ def create_app(config=None):
         app.logger.handlers[0].setFormatter(log_formatter)
         logging.getLogger().handlers[0].setFormatter(log_formatter)
 
+    #Scss is expecting this
+    app.debug = app.config.get('DEBUG')
+    scss_compiler = Scss(app, static_dir='static/css', asset_dir='static/scss')
+    scss_compiler.update_scss()
```
Comment (Matir) [inline scoreboard/app.py]: Can't do this, as it will not run on AppEngine.

Comment (zwade) [inline scoreboard/app.py]: What specifically won't work. The debug portion, or the entire compilation process?

Comment (k9delight) [inline scoreboard/app.py]: Appengine filesystem is read only. Entire compilation process won't work as
a result. Appengine will need a pre-compilation step before deploy

File: scoreboard/logger.py (hunk 1 above)
Comment (zwade) [inline scoreboard/logger.py]: Addressed by #36

**Comment by zwade** [comment]
appengine compilation delegated to #37
